src/backends/pyplot.jl

```diff
@@ -1308,9 +1308,10 @@ function py_add_legend(plt::Plot, sp::Subplot, ax)
                             linestyle = py_linestyle(series[:seriestype], get_linestyle(series))
                         )
                     elseif series[:seriestype] in (:path, :straightline, :scatter)
+                        hasline = get_linewidth(series) > 0
                         PyPlot.plt."Line2D"((0,1),(0,0),
                             color = py_color(single_color(get_linecolor(series, clims)), get_linealpha(series)),
-                            linewidth = py_thickness_scale(plt, sp[:legendfontsize] / 8),
+                            linewidth = py_thickness_scale(plt, hasline * sp[:legendfontsize] / 8),
```

Comment:
Without this we have a line in the legend marker for `scatter(rand(10))`.
`clamp` was used to have limits for linewidths in the legend markers. But since we are now scaling with the legendfont instead of the plot linewidth this is no longer necessary.

Reply:
ah right, I thought clamp was used to restrict the length of the line